### PR TITLE
f-rating@v0.17.0 - Fix rating star rating state and remove stars flickering

### PR DESCRIPTION
**Pull request by juan-amor**
### Added
- Added `hasRatingData` property to hide the RatingMultiStarVariant component until we have data
- Added mounted on the RatingMultiStarVariant and using Vue.nextTick to set the `hasRatingData` property after Vue.nextTick to ensure the % of masking over the filled stars occurs after the DOM has been updated
- Default to 0 and log a warning when the star rating is not within the allowed range

### Changed
- `starRating` prop default and set it to use `0`.

### Removed
- Props validation for the star rating as this is not stateless

---

## UI Review Checks

- [X] README and/or UI Documentation has been [created|updated]
- [X] Unit tests have been [created|updated]
- [ ] This code has been checked with regard to [our accessibility standards](http://fozzie.just-eat.com/documentation/general/accessibility/checklist)

## Browsers Tested

- [ X Chrome (latest)
- [ ] Mobile (Please list device/browser – Ideally one iPhone model and one Android)

### List any other browsers that this PR has been tested in:

- [View the Browser Support Checklist](http://fozzie.just-eat.com/documentation/general/browser-support)

### CODE REVIEW

File: packages/components/molecules/f-rating/src/components/RatingMultiStarVariant.vue [hunk 1]
```diff
@@ -13,6 +13,7 @@
         </div>
 
         <div
+            v-if="hasData"
```
Comment (kevinrodrigues) [inline packages/components/molecules/f-rating/src/components/RatingMultiStarVariant.vue]: Small one but could this be named `hasRatingData`?

Comment (juan-amor) [reply packages/components/molecules/f-rating/src/components/RatingMultiStarVariant.vue]: Done

**Review by kevinrodrigues — DISMISSED** [review]

**Review by kevinrodrigues — DISMISSED** [review]

File: packages/components/molecules/f-rating/src/components/Rating.vue [hunk 2]
```diff
@@ -165,6 +164,20 @@ export default {
...
         shouldDisplayUserOwnRating () {
             return this.isUserRating && this.hasRatingAvailable;
+        },
+
+        /**
+         * Returns the star rating is within the valid range or 0 otherwise
+         *
+         * @returns {boolean}
+         */
+        starRatingWithValidation () {
```
Comment (ashleynolan) [inline packages/components/molecules/f-rating/src/components/Rating.vue]: I'd probably call this:

```suggestion
        validatedStarRating () {
```

(Would need to update the other references to this throughout the PR)

Comment (juan-amor) [reply packages/components/molecules/f-rating/src/components/Rating.vue]: cool, have renamed @ashleynolan , thanks for the suggestion

**Review by kevinrodrigues — APPROVED** [review]

**Pull request merged**